### PR TITLE
Hot reload for static files under webroot

### PR DESCRIPTION
**Pull request by TheAngryByrd**
## Proposed Changes

This adds hot reloading of files under the webroot provided by the `IHostingEnvironment`.  It uses the `IFileProvider` [Watch](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.fileproviders.ifileprovider.watch?view=aspnetcore-2.2) to find changes and notifies the websocket to refresh when changes are detected.  Currently relying on `dotnet watch msbuild /t:CopyFilesTarget` to move files to the correct directory on change.

## Types of changes

What types of changes does your code introduce to Giraffe.HotReload?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] Build and tests pass locally
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added necessary documentation (if appropriate)

## Further comments

So this doesn't cover the cases of people using `UseStaticFiles` and setting different FileProviders such as:

```csharp
 app.UseStaticFiles(new StaticFileOptions
    {
        FileProvider = new PhysicalFileProvider(
            Path.Combine(Directory.GetCurrentDirectory(), "MyStaticFiles")),
        RequestPath = "/StaticFiles"
    });
```

The only way I can think to do this is to allow for the `Settings` to take in an additional list of file paths from the caller.

### CODE REVIEW

**Comment by TheAngryByrd** [comment]
@baronfel should be ready for review

File: samples/ReloadSample/Program.fs [hunk 1]
```diff
@@ -120,13 +123,13 @@ let configureLogging (builder : ILoggingBuilder) =
...
 [<EntryPoint>]
 let main _ =
-    let contentRoot = Directory.GetCurrentDirectory()
-    let webRoot     = Path.Combine(contentRoot, "WebRoot")
+    // let contentRoot = Directory.GetCurrentDirectory()
+    let webroot     = "wwwroot"
     WebHostBuilder()
         .UseKestrel()
-        .UseContentRoot(contentRoot)
+        // .UseContentRoot(contentRoot)
```
Comment (TheAngryByrd) [inline samples/ReloadSample/Program.fs]: Setting the ContentRoot to `Directory.GetCurrentDirectory()` will end up setting it based on working directory.  I don't think this is intended.  Should we just remove this?

File: src/Giraffe.HotReload/Library.fs [hunk 2]
```diff
@@ -206,28 +212,46 @@ module rec LiveUpdate =
...
               logger.LogInformation("Notifying websocket {id}", i)
-              do! socket.SendAsync(refreshCommand, Net.WebSockets.WebSocketMessageType.Text, true, Threading.CancellationToken.None)
+              do! sendRefreshCommand socket
             }
           )
           |> System.Threading.Tasks.Task.WhenAll
           |> ignore
 
+        let handleFileChange () =
+          logger.LogDebug("Static content file changed. Sending refresh commands")
```
Comment (TheAngryByrd) [inline src/Giraffe.HotReload/Library.fs:234]: Unfortunately with the `fileProvider.Watch` it doesn't give feedback as to which file changed. There are many issues like https://github.com/aspnet/Extensions/issues/846 opened around the many repos.

File: samples/ReloadSample/Program.fs (hunk 1 above)
Comment (baronfel) [reply samples/ReloadSample/Program.fs]:
```suggestion
```

yep, we can remove it. the content root is the project directory by default, and the webroot is `projectdir/wwwroot` by default as well, so technically we don't have to do either of these?

File: samples/ReloadSample/Program.fs [hunk 3]
```diff
@@ -120,13 +123,13 @@ let configureLogging (builder : ILoggingBuilder) =
 
 [<EntryPoint>]
 let main _ =
-    let contentRoot = Directory.GetCurrentDirectory()
-    let webRoot     = Path.Combine(contentRoot, "WebRoot")
+    // let contentRoot = Directory.GetCurrentDirectory()
```
Comment (baronfel) [inline samples/ReloadSample/Program.fs:126]:
```suggestion
```

File: samples/ReloadSample/Program.fs [hunk 4]
```diff
@@ -120,13 +123,13 @@ let configureLogging (builder : ILoggingBuilder) =
...
+    let webroot     = "wwwroot"
```
Comment (baronfel) [inline samples/ReloadSample/Program.fs:127]:
```suggestion
```

File: samples/ReloadSample/Program.fs [hunk 5]
```diff
@@ -120,13 +123,13 @@ let configureLogging (builder : ILoggingBuilder) =
...
     WebHostBuilder()
         .UseKestrel()
-        .UseContentRoot(contentRoot)
+        // .UseContentRoot(contentRoot)
         .UseIISIntegration()
-        .UseWebRoot(webRoot)
+        .UseWebRoot(webroot)
```
Comment (baronfel) [inline samples/ReloadSample/Program.fs:131]:
```suggestion
```

**Comment by baronfel** [comment]
I made suggestions to remove the unnecessary overrides of the default contentroot/webroot props, but other than that this is good to go!

File: samples/ReloadSample/Program.fs (hunk 1 above)
Comment (TheAngryByrd) [reply samples/ReloadSample/Program.fs]: I figured having the webroot being explicit would be easier for people to figure out how to change things if needed.

**Comment by baronfel** [comment]
Thanks!

**Comment by baronfel** [comment]
I'll push a new version sometime today with this included.

**Pull request merged**